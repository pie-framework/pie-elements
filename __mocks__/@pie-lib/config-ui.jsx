import React from 'react';

const Comp = (name) => (props) => <div compName={name}>{name}</div>;

module.exports = {
  NumberTextField: Comp('NumberTextField'),
  InputRadio: Comp('InputRadio'),
  InputCheckbox: Comp('InputCheckbox'),
  TwoChoice: Comp('TwoChoice'),
  InputContainer: Comp('InputContainer'),
  FeedbackConfig: Comp('FeedbackConfig'),
  Tabs: Comp('Tabs'),
  Checkbox: Comp('Checkbox'),
  FeedbackSelector: Comp('FeedbackSelector'),
  choiceUtils: {
    firstAvailableIndex: jest.fn(() => '1'),
  },
};
