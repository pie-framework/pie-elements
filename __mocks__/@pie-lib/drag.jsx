
export const uid = {
  withUid: jest.fn(input => input),
  generateUid: jest.fn().mockReturnValue('1'),
  Provider: jest.fn(c => c)
};

export const PlaceHolder = props => (
  <div componentName="placeholder">{props.children}</div>
);

export const withDragContext = jest.fn(i => i);

export const swap = (input) => input
