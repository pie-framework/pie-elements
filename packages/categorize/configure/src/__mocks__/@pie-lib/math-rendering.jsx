import React from 'react';

export const renderMath = jest.fn();
export const HtmlAndMath = (props) => <div>{props.html}</div>;
