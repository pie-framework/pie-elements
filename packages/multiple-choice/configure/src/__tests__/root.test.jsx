import { shallow } from 'enzyme';
import React from 'react';

import Root from '../root';
import { choiceUtils as utils } from '@pie-lib/config-ui';

const model = () => ({
  prompt: 'Which of these northern European countries are EU members?',
  choiceMode: 'checkbox',
  keyMode: 'numbers',
  choices: [
    {
      correct: true,
      value: 'sweden',
      label: 'Sweden',
      feedback: {
        type: 'none',
        value: ''
      }
    },
    {
      value: 'iceland',
      label: 'Iceland',
      feedback: {
        type: 'none',
        value: ''
      }
    },
    {
      value: 'norway',
      label: 'Norway',
      feedback: {
        type: 'none',
        value:
          ''
      }
    },
    {
      correct: true,
      value: 'finland',
      label: 'Finland',
      feedback: {
        type: 'none',
        value: ''
      }
    }
  ],
  partialScoring: false,
  partialScoringLabel: `Each correct response that is correctly checked and each incorrect response
          that is correctly unchecked will be worth 1 point.
          The maximum points is the total number of answer choices.`,
  configure: {
    /**
     * These are the configurable options
     */
    /*promptLabel : 'Item Stem',
    responseTypeLabel: 'Type of Answer',
    choicesLabel: 'Answer Choice',
    addChoiceButtonLabel: 'Add New Choice',
    enableSelectChoiceLabel: false,
    enableSelectResponseType: false,
    enableAddChoice: false,
    enableAddFeedBack: false,
    enableDeleteChoice: false*/
  }
});

describe('Root', () => {
  let w;
  let onModelChanged = jest.fn();
  let initialModel = model();

  const wrapper = extras => {
    const defaults = {
      onModelChanged,
      model: model()
    };
    const props = { ...defaults, ...extras };

    return shallow(<Root {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('onChoiceModeChanged', () => {
      it('resets the model', () => {
        w = wrapper();
        w.instance().onChoiceModeChanged('radio');

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ choiceMode: 'radio' }),
          true
        );
      });
    });

    describe('onRemoveChoice', () => {
      it('removes choice', () => {
        w = wrapper();
        w.instance().onRemoveChoice(0);

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ choices: initialModel.choices.slice(1) }),
          undefined
        );
      });
    });

    describe('onPartialScoringChanged', () => {
      it('changes partial scoring value', () => {
        w = wrapper();
        w.instance().onPartialScoringChanged(true);

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ partialScoring: true }),
          undefined
        );
      });
    });

    describe('onAddChoice', () => {
      it('adds a choice', () => {
        w = wrapper();
        w.instance().onAddChoice();

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ choices: [
            ...initialModel.choices,
            {
              label: 'label',
              value: utils.firstAvailableIndex(initialModel.choices.map(c => c.value), 0),
              feedback: {
                type: 'none'
              }
            }
          ] }),
          undefined
        );
      });
    });

    describe('onKeyModeChanged', () => {
      it('changes keyMode', () => {
        w = wrapper();
        w.instance().onKeyModeChanged('letters');

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ keyMode: 'letters' }),
          undefined
        );
      });
    });

    describe('onChoiceChanged', () => {
      it('changes choice', () => {
        let choice = {
          correct: true,
          value: 'iceland',
          label: 'Iceland',
          feedback: {
            type: 'none',
            value: ''
          }
        };

        w = wrapper();
        w.instance().onChoiceChanged(1, choice);
        initialModel.choices.splice(1, 1, choice);

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ choices: initialModel.choices }),
          undefined
        );
      });

      it('changes choice and makes incorrect all other choices', () => {
        let choice = {
          correct: true,
          value: 'iceland',
          label: 'Iceland',
          feedback: {
            type: 'none',
            value: ''
          }
        };
        let instance;

        w = wrapper();
        instance = w.instance();

        instance.onChoiceModeChanged('radio');
        instance.onChoiceChanged(1, choice);

        initialModel.choices.splice(1, 1, choice);

        expect(onModelChanged).toBeCalledWith(
          expect.objectContaining({ choices: [
            {
              correct: false,
              value: 'sweden',
              label: 'Sweden',
              feedback: {
                type: 'none',
                value: ''
              }
            },
            {
              correct: true,
              value: 'iceland',
              label: 'Iceland',
              feedback: {
                type: 'none',
                value: ''
              }
            },
            {
              correct: false,
              value: 'norway',
              label: 'Norway',
              feedback: {
                type: 'none',
                value:
                  ''
              }
            },
            {
              correct: false,
              value: 'finland',
              label: 'Finland',
              feedback: {
                type: 'none',
                value: ''
              }
            }
          ] }),
          undefined
        );
      });
    });
  });
});

