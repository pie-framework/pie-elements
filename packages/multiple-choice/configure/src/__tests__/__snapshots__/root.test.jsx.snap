// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root snapshot renders 1`] = `
<WithStyles(Main)
  model={
    Object {
      "choiceMode": "checkbox",
      "choices": Array [
        Object {
          "correct": true,
          "feedback": Object {
            "type": "none",
            "value": "",
          },
          "label": "Sweden",
          "value": "sweden",
        },
        Object {
          "feedback": Object {
            "type": "none",
            "value": "",
          },
          "label": "Iceland",
          "value": "iceland",
        },
        Object {
          "feedback": Object {
            "type": "none",
            "value": "",
          },
          "label": "Norway",
          "value": "norway",
        },
        Object {
          "correct": true,
          "feedback": Object {
            "type": "none",
            "value": "",
          },
          "label": "Finland",
          "value": "finland",
        },
      ],
      "configure": Object {},
      "keyMode": "numbers",
      "partialScoring": false,
      "partialScoringLabel": "Each correct response that is correctly checked and each incorrect response
          that is correctly unchecked will be worth 1 point.
          The maximum points is the total number of answer choices.",
      "prompt": "Which of these northern European countries are EU members?",
    }
  }
  onAddChoice={[Function]}
  onChoiceChanged={[Function]}
  onChoiceModeChanged={[Function]}
  onKeyModeChanged={[Function]}
  onPartialScoringChanged={[Function]}
  onPromptChanged={[Function]}
  onRemoveChoice={[Function]}
/>
`;
